language/changes/5-named-addresses.md

```diff
@@ -0,0 +1,212 @@
+# Named Addresses
+
+- Status: Implemented in Move 1.3
+
+## Introduction
+
+Named addresses are a new source language only feature (that is, the named address feature is
+compiled away and not present in the Move bytecode format). The feature allows names to be used in
+place of numerical values in any spot where addresses are used. Named addresses are declared as top
+level elements (outside of modules and scripts) with `address MyAddr;` and can be assigned like so
+`address MyAddr = 0x42;`
+
+With the landing of this feature, Move standard library modules now reside under the address `Std`
+and must be used as such, e.g. `Std::Vector`. Similarly, diem framework modules now reside under the
+address `DiemFramework`, e.g. `DiemFramework::XUS`.
+
+Alongside this change, there is new syntax for address literals when used as expression values, e.g.
+`let addr = @0x42;`.
+
+## Motivations
+
+Fixed, numerical addresses were "good enough" for the language starting off. But the inability to
+set addresses via a config at build time will severely hinder code portability and usability long
+term. Additionally, the lack of named address support has been painful both for account
+configuration in testing and basic readability of code.
+
+To combat this, named addresses can be added as a source language feature. They will still compile
+down to the same address system that exists today, but it will greatly increase the portability,
+testability, and readability of source language programs.
+
+## Description
+
+### New Address Literal Syntax
+
+Addresses now come in two flavors, named or numerical. The syntax for a named address follows the
+same rules for any named identifier in Move. The syntax of a numerical address is no longer
+restricted to `0x`-prefixed values, and now any valid numerical literal can be used.
+
+To make room for the named address feature, address expression values have a new syntax. This new
+syntax reduces the complexity around named addresses as it prevents shadowing issues around module
+members (specifically constants) and local variables.
+
+In the old syntax, all address values began with `0x` and this hex prefix could not be used for
+integer values. In the new syntax, all address values are prefixed with `@`. Following the `@`
+operator any valid address can be used. For example:
+
+```move
+let _: u8 = 0x1u8;
+let _: u64 = 0x42u64;
+let _: u128 = 0x42u128;
+let a1: address = @Std;
+let a2: address = @66;
+let a3: address = @0x42;
+```
+
+In a way, you can think of `@` as an operator that takes an address from being a namespace item to
+an expression item.
+
+### Declaring Named Addresses
+
+In this new system, anywhere numerical address literals are used, a name can be used instead. New
+address names can be declared as:
+
+```move
+address MyAddr;
+```
+
+Using an address block will also declare the name:
```

Comment:
Kind of unrelated, but: is there still a plan to deprecate address blocks?

Reply:
Maybe? They might still be useful